spec/models/comment_spec.rb

```diff
@@ -0,0 +1,20 @@
+# spec/models/comment_spec.rb
+
+require 'rails_helper'
+
+RSpec.describe Comment, type: :model do
+  let(:user) { create(:user, name: 'Supreeti') }
+  let(:post) { create(:post, title: 'Test user') }
+
+  describe 'associations' do
+    it { should belongs_to(:post).with_foreign_key(:post_id) }
+    it { should belongs_to(:user).with_foreign_key(:user_id) }
+  end
+
+  describe 'callbacks' do
+    it 'updates the comments_counter after save' do
+      comment = build(:comment, post: 'Test user', user: 'Supreeti')
+      expect { comment.save }.to change { post.reload.comment_counter }.by(1)
+    end
+  end
```

Comment:
- [ ] I'm getting failure here as well. you can use Factorybot to create a user, post, and comment.

![Screenshot from 2023-10-01 03-44-52](https://github.com/supreeti/BlogApp-ROR/assets/26767383/3c19e11f-8980-47fb-90f4-af5d898cf400)